src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -131,6 +131,8 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
           break;
         case JOURNAL_NODES :
           if (liveState.getJournalNodeSize() == conf.getJournalNodeCount()) {
+            // TODO move the reload to correctCurrentPhase and make it idempotent
+            reloadConfigsOnAllRunningTasks(driver);
```

Comment:
Cool. Just wanted to make sure I understood the scenario. Would love to see this logic go into correctConcurrentPhase so the actions on state transitions are clearer, but that's what the TODO is for. :)